src/services/LoggingService.js

```diff
@@ -12,20 +12,24 @@ class LoggingService {
 
   static logError(error) {
     if (typeof newrelic !== 'undefined') {
+      // Note: customProperties are not sent.  Presumably High-Security Mode is being used.
       newrelic.noticeError(error);
     }
   }
 
+  // Note: will simply log errors that don't seem to be API error responses.
   static logAPIErrorResponse(error) {
-    let { message } = error;
+    let processedError = Object.create(error);
     if (error.response) {
-      message = `${error.response.status} ${error.response.config.url} ${JSON.stringify(error.response.data)}`;
+      processedError = new Error(`API request failed: ${error.response.status} ${error.response.config.url} ${JSON.stringify(error.response.data)}`);
     } else if (error.request) {
-      message = `${error.request.status} ${error.request.responseURL} ${error.request.responseText}`;
-    } else if (error.stack) {
-      message = error.stack;
+      processedError = new Error(`API request failed: ${error.request.status} ${error.request.responseURL} ${error.request.responseText}`);
     }
-    this.logError(new Error(`API request failed: ${message}`));
+    if (processedError.message) {
+      // NewRelic will not log the error if it is too long.
+      processedError.message = processedError.message.substring(0, 4000);
```

Comment:
FYI: We had places where `error.response.data` was an entire HTML 500 page, and the message got so long that nothing was sent to NewRelic and the whole things was lost.  This limit fixes that issue.
FYI: @georgebabey

Reply:
Good catch - thanks @robrap

Reply:
Would be nice to make that a well-named constant.

Reply:
@albemarle: Agreed and good point.  Will add that to the refactor.  Thanks.